src/components/shared/inputs/input-shell.jsx

```diff
@@ -0,0 +1,49 @@
+import clsx from 'clsx';
+import styles from './shared-inputs.module.scss';
+
+export default function InputShell({
+  type = '',
+  name,
+  label,
+  errors = '',
+  register,
+  classNames = '',
+  required = false,
+}) {
+  const inputType = {
```

Comment:
Nice!! This object is called a hash map, you will often see it on code challenges 👀

Often with hash maps you want the variable name to communicate what is being mapped, something like `mapInputTypeToComponent`.